domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaygroup/DisplayGroupDateTimeDisplayContextParameterValidatorImpl.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.displaygroup;
+
+import com.google.common.base.Strings;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter.DisplayContextParameterType;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.displaycontextparameter.AbstractDisplayContextParameterValidator;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.displaycontextparameter.DisplayContextParameterValidatorFactory;
+import uk.gov.hmcts.ccd.definition.store.repository.FieldTypeUtils;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.DisplayGroupCaseFieldEntity;
+
+import java.util.Optional;
+
+@Component
+public class DisplayGroupDateTimeDisplayContextParameterValidatorImpl extends AbstractDisplayContextParameterValidator<DisplayGroupCaseFieldEntity> implements DisplayGroupCaseFieldValidator {
+
+    private static final DisplayContextParameterType[] ALLOWED_TYPES =
+        { DisplayContextParameterType.DATETIMEDISPLAY };
+    private static final String[] ALLOWED_FIELD_TYPES =
+        { FieldTypeUtils.BASE_DATE, FieldTypeUtils.BASE_DATE_TIME };
+
+    public DisplayGroupDateTimeDisplayContextParameterValidatorImpl(DisplayContextParameterValidatorFactory displayContextParameterValidatorFactory) {
+        super(displayContextParameterValidatorFactory, ALLOWED_TYPES, ALLOWED_FIELD_TYPES);
+    }
+
+    @Override
+    public ValidationResult validate(DisplayGroupCaseFieldEntity entity) {
+        return shouldSkipValidatorForEntity(entity) ? new ValidationResult() : super.validate(entity);
+    }
+
+    @Override
+    protected String getDisplayContextParameter(DisplayGroupCaseFieldEntity entity) {
+        return entity.getDisplayContextParameter();
+    }
+
+    @Override
+    protected String getFieldType(DisplayGroupCaseFieldEntity entity) {
+        return entity.getCaseField().getBaseTypeString();
+    }
+
+    private boolean shouldSkipValidatorForEntity(DisplayGroupCaseFieldEntity entity) {
+        final String displayContextParameter = getDisplayContextParameter(entity);
+        if (!Strings.isNullOrEmpty(displayContextParameter)) {
+            final Optional<DisplayContextParameterType> parameterType =
+                DisplayContextParameterType.getParameterTypeFor(displayContextParameter);
```

Comment:
what if we added a method to `DisplayGroupCaseFieldEntity` that returns a displayContextParameter as a `DisplayContextParameter` other than String? would this remove some duplication?

Reply:
At the moment wouldn't reduce much as it would be other entities too, and `domain` is not accessible to `repository` in the current config.